### PR TITLE
make sure order column starts from 0 in group modifications list

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java [hunk 1]
```diff
@@ -39,8 +39,14 @@ public ModificationGroupEntity(UUID uuid) {
     }
 
     public void addModification(ModificationEntity modification) {
-        modifications.add(modification);
-        modification.setGroup(this);
+        if (modifications.isEmpty()) {
+            List<ModificationEntity> newList = new ArrayList<>();
+            newList.add(modification);
+            setModifications(newList);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:45]: `setModifications(List.of(modification)) `?

Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:45]: This was my first shot. But I we pass an unmutable collection like List(), we got an exception when we try to add a second element to the modifications list.

File: src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java [hunk 2]
```diff
@@ -39,8 +39,14 @@ public ModificationGroupEntity(UUID uuid) {
     }
 
     public void addModification(ModificationEntity modification) {
-        modifications.add(modification);
-        modification.setGroup(this);
+        if (modifications.isEmpty()) {
```
Comment (TheMaskedTurtle) [inline src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:42]: Maybe add a comment to explain we have to do that to maintain proper order with @OrderColumn annotation

**Review by TheMaskedTurtle — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:45]: Arrays.asList

Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:45]: Arrays.asList also returns "a fixed-size list"

Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:45]: same issue with:
 List<ModificationEntity> newList = Arrays.asList(modification);

Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:45]: sonar doesnt like it ;-)
_Double Brace Initialization should not be used_

File: src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java (hunk 2 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:42]: done

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**